contracts/example/tests/test.rs

```diff
@@ -52,15 +52,16 @@ fn setup_app<'a>(
 #[test]
 fn test_gmp_example() {
     let env = Env::default();
-
     env.mock_all_auths();
```

Comment:
remove auth mocking so that the auth can be verified

Reply:
I tried to remove `env.mock_all_auths();`. Without having it, Token mint would fail. 
We want to test a flow of gmp call here, not the mint function. so I'd rather keep it here.

```
HostError: Error(Auth, InvalidAction)

Event log (newest first):
   0: [Diagnostic Event] topics:[error, Error(Auth, InvalidAction)], data:"escalating error to panic"
   1: [Diagnostic Event] topics:[error, Error(Auth, InvalidAction)], data:["contract call failed", mint, [CAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD2KM, 100]]
   2: [Failed Diagnostic Event (not emitted)] contract:CBEPDNVYXQGWB5YUBXKJWYJA7OXTZW5LFLNO5JRRGE6Z6C5OSUZPCCEL, topics:[error, Error(Auth, InvalidAction)], data:["Unauthorized function call for address", CAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD2KM]
   3: [Diagnostic Event] topics:[fn_call, Bytes(48f1b6b8bc0d60f7140dd49b6120fbaf3cdbab2adaeea631313d9f0bae9532f1), mint], data:[CAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD2KM, 100]

```